### PR TITLE
updating share UI

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
This PR improves sharing functionality by:
 - allowing to create share without basket,
 - chosen plugin in members section
 - syncing share or discussion edit with metadata panel

Note: Discussions are deprecated now. They still will be supported in the display but new one cannot be created. Basket was removed in favor of Create Share button.

To test:
 - create share and discussion.
 - edit share and discussion
 - test if data saved correctly

cc: @pwalczysko

### CODE REVIEW

**Comment by pwalczysko** [comment]
Testing Share:

   - in the UI, this all works as expected, the user selection dialog is much improved, the editing of the Share is respected and updated
   - sending the emails: I am not sure what the expected behaviour is in share edits : I edited twice, first time just added one more member to the share, second time edited the title=message of the share. I think that the other members of the share obtained email only after the first edit (with the same message as before, not saying anything about why this new message have been produced). The second time (when the message was changed) no email came. This seems not logical, or there is a bug in the email sending when the message is changed.

**Comment by pwalczysko** [comment]
When creating a Discussion the member selection dialog is as it used to be (not very selection friendly). Why is it not the same as on the Share ( = the nicely usable one) ?
screen shot 2015-05-14 at 18 58 13

**Comment by pwalczysko** [comment]
When creating Discussion, I have forgotten to add a message. Instead of being stopped after I clicked Accept, the dialog vanished and I was left wondering what happened - it seems like some regression, previously I would not be able to click Accept without a message.

**Comment by pwalczysko** [comment]
Bug: Now even the Share has the old bad menu ...Obviously, if you first choose Discussion (with the bad menu dialog), followed by Share, also the Share will have bad menu, just like Discussion. If you first choose Share, it has the nice good menu. Some sort of cashing in the Public tab ?

**Comment by will-moore** [comment]
This PR is causing a bunch of failures in "Test Basket Share" and "Test Delete Images in Share". See:
https://ci.openmicroscopy.org/view/Failing/job/OMERO-5.1-merge-robotframework/242/robot/

If you could fix "Test Basket Share" in this PR, then I can apply the same changes to "Test Delete Images in Share" in #3803.

**Comment by aleksandra-tarkowska** [comment]
@pwalczysko I cleand up some templates and now should be better. Please retest on Monday.

@will-moore sorry about not altering robot tests and thanks for helping with them. Now they should pass

```
web-chrome:
==============================================================================
New Test :: Tests submission of forms.                                        
==============================================================================
Test Basket Share :: Tests adding image to basket, creating a shar... | PASS |
------------------------------------------------------------------------------
New Test :: Tests submission of forms.                                | PASS |
1 critical test, 1 passed, 0 failed
1 test total, 1 passed, 0 failed
==============================================================================

web-chrome:
==============================================================================
New Test :: Tests submission of forms.                                        
==============================================================================
Test Discussion                                                       | PASS |
------------------------------------------------------------------------------
New Test :: Tests submission of forms.                                | PASS |
1 critical test, 1 passed, 0 failed
1 test total, 1 passed, 0 failed
==============================================================================
```

**Comment by will-moore** [comment]
I know that failure at https://ci.openmicroscopy.org/view/Failing/job/OMERO-5.1-merge-robotframework/243/robot/Web%20&%20Web/Web/Forms%20Test/Test%20Basket%20Share/ isn't caused by this PR, but since you're currently editing this test, I think a small sleep either side of the ```Click Element     basketButton``` will help with this issue as I've done in https://github.com/will-moore/openmicroscopy/commit/643013796446423f85478b1677842aa027e62036
Otherwise I can add in a later PR if this one is being merged today?

**Comment by pwalczysko** [comment]
@aleksandra-tarkowska : the form is okay now (with respect to user selection in Discussion)
What is still there is the

  - problem with being able to click Accept even when the form is incomplete (missing message) see https://github.com/openmicroscopy/openmicroscopy/pull/3813#issuecomment-102119700
 
Workflow: 
  - create a Discussion
  - in the form do NOT put any message into the box, just select some members
  - click Accept button
  - the form vanishes, no discussion was created, no warning produced -> not expected

Before htting Accept:

screen shot 2015-05-18 at 10 16 02

After hitting Accept

screen shot 2015-05-18 at 10 16 14

**Comment by pwalczysko** [comment]
Unfortunately, now the Share is with the old form ! - this probably happens because I have hit the https://github.com/openmicroscopy/openmicroscopy/pull/3813#issuecomment-102987954 problem ?

**Comment by joshmoore** [comment]
@aleksandra-tarkowska : when you have time to push a commit, you can just remove the `--exclude` in the description.

**Comment by joshmoore** [comment]
@aleksandra-tarkowska : should we keep this on the radar for 5.1.3?

**Comment by aleksandra-tarkowska** [comment]
I will look at this next week as I mostly cleaned out registry now. Waiting on CI to be back to deploy

**Comment by mtbc** [comment]
CI is back, I thought?

**Comment by aleksandra-tarkowska** [comment]
```
==============================================================================
Test Share :: Tests adding image to basket, creating a share and e... | PASS |
------------------------------------------------------------------------------
==============================================================================
Test Delete Images in Share :: Deletes pre-imported images after p... | PASS |
------------------------------------------------------------------------------
```

**Comment by aleksandra-tarkowska** [comment]
With all 3 external apps plugged tests now are passing

```
test/integration/test_csrf.py .........................

=========================================================== 25 passed in 41.20 seconds ===========================================================
```

**Comment by aleksandra-tarkowska** [comment]
@pwalczysko is going to test it on breaking

**Comment by pwalczysko** [comment]
Tested on breaking.
In general, happy about the removal of basket. 
The new workflow is much easier.

The new UI is much easier to use.
Couple of points:

1. when Editing existing share, tooltip is not very helpful on the dialog (says just ``Edit Share``), either remove this tooltip or put more granular content

screen shot 2015-06-25 at 15 57 24
2. when disabling the Share, it might be useful to highlight the Share as disabled in the left-hand tree (also for the owner of the Share). The ``Disabled`` text which is there for the other users (non-owners of the share) might do the trick as well

3. in the share dialog, the "helping question marks" do not do anything (no tooltip or popup)

4. make possible also to Delete shares (as discussed, this one would be for another PR)

I suppose we could live without the Discussion feature ? (was removed in this PR apparently)
cc @gusferguson

**Comment by gusferguson** [comment]
I can live without the discussion.
If this makes it into 5.1.3 it will need a Help update. 
@aleksandra-tarkowska - how likely is it to make the release?

**Comment by aleksandra-tarkowska** [comment]
I will fix @pwalczysko comments and if tests passes it is ready to go in.

**Comment by aleksandra-tarkowska** [comment]
@gusferguson it is on breaking because these are breaking changes and broke the build.

**Comment by joshmoore** [comment]
(Taking out of breaking for Monday's build)

**Comment by aleksandra-tarkowska** [comment]
@pwalczysko these should address above issues

**Comment by pwalczysko** [comment]
Bug: When you try to disable (or enable) a Share, it tends to try to update the first Share you edited instead of the one you are editing now (some unfortunate cashing is going on if you edit 2 or more Shares one after each other).
Workflow:

  - select a Share Share 21752 on trout merge user-4
  - edit this share and enable (or disable) it
  - hit Accept and observe that the change was respected
  - now select another Share , e.g. Share 21749  of the same user
  - enable or disable it as appropriate and hit Accept
  - the Share Share 21749 will NOT have any changes, instead
  - the Share 21752 WILL have the changes

**Comment by pwalczysko** [comment]
On the Markups tooltip, I would reword: ``Markups: If you'd like to include URL, please enter it in the format : http:/.......py.org/

Other than that, the tooltips are fine.

**Comment by pwalczysko** [comment]
Markup of Expired Share: See for example Share Share 17602 - this one is expired. 

1. In the right-hand pane it says ACTIVE EXPIRED -> although this might make sense programatically, in english it is not possible (what is expired cannot be ACTIVE) -  Suggest to remove the word ACTIVE, leave just EXPIRED in this case

2. Mark in the left-hand tree the EXPIRED light grey (the same as NOT ACTIVE) - principally I would think thet the NOT ACTIVE and EXPIRED is more or less the same state, although reached by two different routes

**Comment by pwalczysko** [comment]
3 problems (out of which 1 bug, and 2 RFEs) still persisting.

**Comment by aleksandra-tarkowska** [comment]
@pwalczysko thanks I will check that

**Comment by gusferguson** [comment]
@aleksandra-tarkowska 

Tested using trout merge user-4 and user-2 read-only-1

The updating after editing a share is working as expected (bug resolved).
The EXPIRED label in the RHP is as expected (i.e. ACTIVE removed).
The EXPIRED and INACTIVE shares both have greyed out icons.

The greying out/restoring of icons after a change in status of the share are not taking place immediately - share icon remains greyed out after share restored to active and vice versa - see screenshots 1 and 2. Requires refresh to update icons. Would be good if this could be done.

A URL inserted into the message field using the format advised in the tooltip does not become an active link in the right hand pane - see screenshot 3. If this is meant to be the case there is no point to the tooltip?

safari

safari


safari

**Comment by aleksandra-tarkowska** [comment]
@gusferguson it is true that we escape url and that is no longer valid. It has been like that for a while. Anyway removed now. I also fixed icon update. I presume that should resolve all the issues you noticed.

**Comment by gusferguson** [comment]
@aleksandra-tarkowska - that sounds great.

**Comment by gusferguson** [comment]
@aleksandra-tarkowska 

Tested using trout merge user-3 read-only-1

No tooltip appearing on message.

**Comment by gusferguson** [comment]
Good to merge.

**Comment by joshmoore** [comment]
Thanks, @gusferguson. @chris-allan went through decorators with a :thumbsup:. Merging.

**Comment by aleksandra-tarkowska** [comment]
@gusferguson ready for you to upgrade help. thanks

**Comment by gusferguson** [comment]
@aleksandra-tarkowska  - thanks - will have it done for the 5.1.3 release.

**Comment by aleksandra-tarkowska** [comment]
@pwalczysko when you are back could you please update doc testing in ome-internal accordingly. thanks
